### ISSUE TITLE
Support sending both c-input and attribute for custom tags

### ISSUE THREAD

**Issue by fierysunset** [raptorjs/marko#53]
`c-input` works well for nested tags, but there are times when we need to pass down extra information from the parent tag, like this:

```
<tagname c-input="data.someObj" attribute="test"/>
```

Examples include when the parent contains the loop and we might need an index in the nested tag, or if parent tag-level logic affects a child tag. 

This should also include support for passing a conditional, or at least a variable (that we can set with a conditional earlier in the file).

Thanks!

**Comment by philidem** [comment]
@fierysunset Is the problem that the `attribute` input value in your example is not available when using `c-input`? Since you can only have one input to the tag (`c-input` takes precedence in this case) I am guessing that `attribute` will not be passed along. Technically, I think the compiler should flag this as an error.

Here are some alternatives:
- Put the value of `attribute` into `data.someObj` and use `<tagname c-input="data.someObj" />`. For example:
```javascript
<assign var="data.someObj.attribute" value="test" />
<tagname c-input="data.someObj" />
```
- Don't use `c-input` but allow wildcard attributes for `tagname` tag. See https://github.com/raptorjs/marko#defining-attributes
- Declare a `data` (or some other name) attribute for `tagname` that could be used as an alternate way to provide input. You could then use it like:
```html
<tagname data="data.someObj" attribute="test"/>
```
Because `data` is not a special attribute, the renderer for `tagname` could access both `data` and `attribute`.

**Comment by patrick-steele-idem** [comment]
I think the desired behavior is for the attributes to be merged into the data object provided by `c-input`. The properties provided via attributes should take precedence over the properties in the the `c-input` object. Also, we should not modify the object provided via `c-input` and instead create a new object with the merged properties.

Is that what you are thinking as well, @fierysunset ? Thoughts @philidem ?

**Comment by patrick-steele-idem** [comment]
Also, there are plenty of workarounds to avoid making this change, but I do think it is something we should just go ahead and fix since the current behavior of silently discarding the attributes when used in conjunction with `c-input` is not right.

**Comment by patrick-steele-idem** [comment]
New version published: `marko@2.3.0`

**Comment by fierysunset** [comment]
Yes to @patrick-steele-idem and thanks!
